app/src/main/java/com/rkm/tasky/feature/loginscreen/screen/LoginScreen.kt

```diff
@@ -0,0 +1,157 @@
+package com.rkm.tasky.feature.loginscreen.screen
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import com.rkm.tasky.R
+import com.rkm.tasky.feature.loginscreen.viewmodel.LoginViewModel
+import com.rkm.tasky.ui.component.EmailValidationTextField
+import com.rkm.tasky.ui.component.PasswordTextField
+
+@Composable
+fun LoginScreen(
+    viewModel: LoginViewModel,
+    modifier: Modifier
+) {
```

Comment:
I'd avoid passing the ViewModel directly to composables, except for the root composable of each screen. This has 3 major reasons:
1. You strongly couple your composables to this specific ViewModel reference. You would not be able to reuse the composable at a place where you don't have a `LoginViewModel`.
2. You're making it harder to have a preview, since you'll need to provide a fake ViewModel instance and fakes for everything the ViewModel needs in its constructor.
3. UI tests are also a bit harder since you'll need to provide these fakes.

What I suggest instead (feel free to use another DI lib like Koin):
```
// In your ViewModel
@HiltViewModel
class SampleViewModel @Inject constructor(): ViewModel() {
    private val _state = MutableStateFlow(SampleState()) // Or Compose state alternatively
    val state = _state.asStateFlow()

    fun onAction(action: SampleAction) {
        when(action) {
            is SampleEvent.Action1 -> // Handle action1
            is SampleEvent.Action2 -> // Handle action2
            // ...
        }
    }

    // ...
}

// In your NavHost
composable(Routes.sample) {
    SampleScreenRoot(navController)
}

// Your screen composable
@Composable
fun SampleScreenRoot(
    navController: NavHostController,
    viewModel: SampleViewModel = hiltViewModel()
) {
    val state by viewModel.state.collectAsStateWithLifecycle()
    SampleScreen(
        state = state,
        onAction = viewModel::onAction
    )
}

@Composable
fun SampleScreen(
    state: SampleState,
    onAction: (SampleAction) -> Unit
) {
    // ...
    Button(onClick = { onAction(SampleAction.ClickedButtonX) } {
        Text("Click me")
    }
}
```
This way, it's easy to preview and test `SampleScreen` since it doesn't depend on a ViewModel reference.